App/Sagas/TextileSagas.js

```diff
@@ -162,11 +167,20 @@ export function * getPhotoHashes ({thread}) {
       try {
         const metasrc = yield call(IPFS.getHashData, hash, '/meta')
         const meta = JSON.parse(Buffer.from(metasrc, 'base64').toString('utf8'))
+        avatarHash = meta.username || meta.peer_id
+        username = meta.username || 'anonymous'
         item = {...item, meta}
       } catch (err) {
         // gracefully return an empty meta for now
       }
-      data.push(item)
+      try {
+        const avatar = avatars[avatarHash] || jdenticon.toSvg(avatarHash, 200)
+        avatars[avatarHash] = avatar
+        item = {...item, avatar}
```

Comment:
You see any good reason to store the svg avatar in the backing data when we can just derive it closer to the UI layer? I'm thinking the other place for it would be `mapStateToProps` in `TextilePhotos`.

Reply:
your call, i can move it

Reply:
Not `TextilePhotos`. I meant `PhotoViewerScreen`. Or even in the render function of the metadata component in there.

Reply:
yeah sure, easy